src/Extensions/TimedScopes/Internal/TimedScopeEventSender.cs

```diff
@@ -0,0 +1,97 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+
+using System;
+using System.Diagnostics;
+using System.Globalization;
+using Microsoft.Extensions.Hosting;
+using Microsoft.Extensions.Logging;
+using Microsoft.Omex.Extensions.Abstractions;
+
+namespace Microsoft.Omex.Extensions.TimedScopes
+{
+	internal sealed class TimedScopeEventSender : ITimedScopeEventSender
+	{
+		public TimedScopeEventSender(TimedScopeEventSource eventSource, IHostEnvironment hostEnvironment, ILogger<TimedScopeEventSender> logger)
+		{
+			m_eventSource = eventSource;
+			m_serviceName = hostEnvironment.ApplicationName;
+			m_logger = logger;
+		}
+
+
+		public void LogTimedScopeEndEvent(TimedScope scope)
+		{
+			if (!m_eventSource.IsEnabled())
+			{
+				return;
+			}
+
+			string serviceName = m_serviceName;
+			string subtype = scope.SubType;
+			string metadata = scope.Metadata;
+			TimedScopeResult result = scope.Result;
+			Activity activity = scope.Activity;
+			string name = activity.OperationName;
+			string correlationId = activity.Id;
+			double durationMs = activity.Duration.TotalMilliseconds;
+			string userHash = activity.GetUserHash(); //TODO: We need add midleware that will set userhash in compliant way and IsTransaction GitHub Issue #166
```

Comment:
Nitpick: typo in middleware

Reply:
I will resolve it in the follow-up pr https://github.com/microsoft/Omex/pull/168